Update Docker assets

labay-adam opened the pull request:
Update the Dockerfile and associated setup script to reflect the changes made in https://github.com/JarodMica/ai-voice-cloning/pull/77
Principal change was the use of pyenv to force Python 3.11.10 in a world when 3.12 is the norm. Updated the base image, which also required changing the default UID/GID.

Commit:
Update Docker assets

JarodMica commented:
Thanks for the PR @labay-adam!

Any reason to move to 12.6.0 and 24.04 ubuntu - is that standard now for ubuntu?

Since I'm on 22.04, I just changed it to `FROM nvidia/cuda:12.2.0-devel-ubuntu22.04` for my build, but I'm also not able to connect to localhost:7860 with my WSL2 rn and I don't feel like debugging what's happening.  Docker built fine and I see no errors in the terminal - I just can't verify the inference/training.  

If you didn't run into any issues, I'll merge this.

labay-adam commented:
I'll muck about a bit more to confirm.

Commit:
Drop activation of pyenv and add Gradio Server

labay-adam commented:
Looks like the issue with the server visibility was the lack of an `GRADIO_SERVER_NAME` env variable. While I was there, though, I cleaned out some misusage of `pyenv` and reverted the base image back to `nvidia/cuda:12.2.0-devel-ubuntu22.04`. Worked a treat on an Arch host, but am happy to do the diligence on some other hosts as needed.

Commit:
Remove venv

Commit:
Change base repo pending merge

Commit:
Lock versions
